test/driver-specs.js

```diff
@@ -0,0 +1,131 @@
+import _ from 'lodash';
+import sinon from 'sinon';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+let expect = chai.expect;
+chai.use(chaiAsPromised);
+
+const BASE_CAPS = {platformName: 'Fake', app: TEST_FAKE_APP};
+
+describe('AppiumDriver', () => {
+  describe('getAppiumRouter', () => {
+    it('should return a route configuring function', async () => {
+      let routeConfiguringFunction = getAppiumRouter({});
+      routeConfiguringFunction.should.be.a.function;
+    });
+  });
+
+  describe('AppiumDriver', () => {
+    function getDriverAndFakeDriver () {
+      let appium = new AppiumDriver({});
+      let fakeDriver = new FakeDriver();
+      let mockFakeDriver = sinon.mock(fakeDriver);
+      appium.getDriverForCaps = function (/*args*/) {
+        return () => {
+          return fakeDriver;
+        };
+      };
+      return [appium, mockFakeDriver];
+    }
+    describe('createSession', () => {
+      let appium
+        , mockFakeDriver;
+      beforeEach(() => {
+        [appium, mockFakeDriver] = getDriverAndFakeDriver();
+      });
+      afterEach(() => {
+        mockFakeDriver.restore();
+        appium.args.defaultCapabilities = {};
+      });
+
+      it('should call inner driver\'s createSession with desired capabilities', async () => {
+        mockFakeDriver.expects("createSession")
+          .once().withExactArgs(BASE_CAPS, undefined, [])
+          .returns([1, BASE_CAPS]);
+        await appium.createSession(BASE_CAPS);
+        mockFakeDriver.verify();
+      });
+      it('should call inner driver\'s createSession with desired and default capabilities', async () => {
+        let defaultCaps = {deviceName: 'Emulator'}
+          , allCaps = _.extend(_.clone(defaultCaps), BASE_CAPS);
+        appium.args.defaultCapabilities = defaultCaps;
+        mockFakeDriver.expects("createSession")
+          .once().withArgs(allCaps)
+          .returns([1, allCaps]);
+        await appium.createSession(BASE_CAPS);
+        mockFakeDriver.verify();
+      });
+      it('should call inner driver\'s createSession with desired and default capabilities without overriding caps', async () => {
+        // a default capability with the same key as a desired capability
+        // should do nothing
+        let defaultCaps = {platformName: 'Ersatz'};
+        appium.args.defaultCapabilities = defaultCaps;
+        mockFakeDriver.expects("createSession")
+          .once().withArgs(BASE_CAPS)
+          .returns([1, BASE_CAPS]);
+        await appium.createSession(BASE_CAPS);
+        mockFakeDriver.verify();
+      });
+    });
+    describe('deleteSession', () => {
+      let appium
+        , mockFakeDriver;
+      beforeEach(() => {
+        [appium, mockFakeDriver] = getDriverAndFakeDriver();
+      });
+      afterEach(() => {
+        mockFakeDriver.restore();
+        appium.args.defaultCapabilities = {};
+      });
+      it('should do nothing if there is no session', async () =>{
```

Comment:
Yes. Both are handled elsewhere. There appears to be no need to unit test `deleteSession` with a non-existent session.